### ISSUE TITLE
`make prep` does not handle NetBSD `gmake`

### ISSUE THREAD

**Issue by SirWumpus** [ioccc-src/mkiocccentry#894]
1. The `gmake prep` generates errors WRT the use of `make`.  It appears the `test_ioccc/prep.sh` is not passed `-m${MAKE}` option and assumes `make`, instead of `$MAKE`.

2. With a some changes (I will push) it will run to completion, but reports these errors
One or more tests failed:

        make_action 25: gmake -f ./Makefile shellcheck: non-zero exit code: 2
        make_action 26: gmake -f ./Makefile seqcexit: non-zero exit code: 2
        make_action 27: gmake -f ./Makefile picky: non-zero exit code: 2

Where the 3 above tools are not installed.

@landon please assign to me as this is a subtask of #893.

**Comment by xexyl** [comment]
Looks like you don't have those tools installed. 

Do you? If not that's probably why. But I can't check for the error codes until tomorrow .. bedtime. 

The Makefile will tell you where to get those tools btw.

**Comment by xexyl** [comment]
Ah you said that already. 

So that's another concern of whether it will work on a workflow. But I don't know. 

I can look at it more tomorrow. If you give more details on the problem maybe that will help too. 

But maybe that's all nonsense: can barely see. Anyway fwiw the Makefile tells where to get the tools. 

Some of them might not be strictly necessary but I don't know. 

Anyway good night!

**Comment by lcn2** [comment]
BTW: Platforms such as **NetBSD** are not the target platforms for `make prep` nor `make release`.  We do **NOT** possess those platforms, nor do we plan to acquire them to be able to support them under  `make prep` nor `make release`.

We have nothing against **NetBSD**, nor **OpenBSD**, nor classic BSD.  As mentioned before, Landon was an early BSD developer when BSD was at Berkeley.

**NOTE**: This does not mean we don't want `make clobber all` to work on as many platforms that is reasonable.  We would want `make clobber all` to work on **NetBSD**, should doing so not negatively impact other platforms.  

If you wish to consider creating a pull request that would help **NetBSD**, and that does not negatively impact our existing support platforms, feel free to suggest one.

**Comment by lcn2** [comment]
One approach we may wish to consider is to better specify the system requirements for running `make prep` and `make release`.  Perhaps in an FAQ?

While there may be workarounds for systems with a shell or make that is less functional than we require: there is also a limit to how far we need to go to accommodate various platforms.

For example, someone with a platform that lacks `bison(1)` or `yacc(1)` won't be successful in running `make release`.   

And for example, someone with a v7 shell 🐚 won't make much success running the shell scripts.

Someone with a Windows development environment might find that `make all` fails, let alone `make prep` and `make release`.

**Comment by lcn2** [comment]
We initially misunderstood and misread the [initial comment](https://github.com/ioccc-src/mkiocccentry/issues/894).  Sorry (tm Canada 🇨🇦).

The development environment requires some tools to be installed.  Without `shellcheck(1)`, `seqcexit(1)` and `picky(1)` `make prep` and `make release` won't work.

We don't want `make prep` or `make release` to proceed unless those tools are able to perform their actions.

This is by design.

If you run `make shellcheck`, or `make seqcexit`, or `make picky` (and your `make(1)` and shell are up to scratch), and the rule correctly detects you lack those required development tools, an error message would have directed you to a location under which those tools may be obtained.

We also suspect that the log files generated by the `make prep` or `make release` would also contain the same error messages with URLs as to where the required tools may be found.

## FYI

The development environment for this repo is not intended to be widely portable.   Only macOS and certain versions of Linux are supported as development environments.  If it happens to work for other environment, fine: but a universal development environment for this repo is not a goal nor was it a requirement.

Remember that it is not required to run `make prep` or `make release` in order to compile `mkiocccentry(1)` and friends.  One does not need to run `make prep` or `make release` in order to use the tools needed to package a submission.

## In conclusion.

Given the above, and given those test failures are by design when the required development tools are missing, we must respectfully close this issue as a "won't fix".

**Comment by lcn2** [comment]
> The gmake prep generates errors WRT the use of make. It appears the test_ioccc/prep.sh is not passed -m${MAKE} option and assumes make, instead of $MAKE.

We would be happy to consider a pull request in regards to the above issue that you reported.

Feel free to submit a pull request to address that issue.

**Comment by SirWumpus** [comment]
OK.  For NetBSD  

* `pkgin install shellcheck`  - this particular tool I'm unclear as to its purpose and wonder if its overkill.
* Installing `seqcexit` from the git repo is straight forward.
* Installing `picky` from git repo is straight forward

`gmake prep` now works and pass...
```
elf$ gmake prep
make_action 10 clobber OK
make_action 11 all OK
make_action 12 depend OK
make_action 13 clean_mkchk_sem OK
make_action 14 all_sem_ref OK
make_action 15 mkchk_sem OK
make_action 16 all OK
make_action 18 parser OK
make_action 19 all OK
make_action 20 load_json_ref OK
make_action 21 use_json_ref OK
make_action 22 clean_generated_obj OK
make_action 23 all OK
make_action 24 bug_report-txl -L bug-report.20240703.103542.txt OK
make_action 25 shellcheck OK
make_action 26 seqcexit OK
make_action 27 picky OK
make_action 28 tags OK
make_action 29 check_man OK
make_action 30 all OK
make_action 31 test OK
All tests PASSED.

make prep: see build.log for details
elf$
```

Does the prep target test the other tools or just the build chain?

**Comment by SirWumpus** [comment]
Installing the missing tools resolves the issue.  There is no change required.
